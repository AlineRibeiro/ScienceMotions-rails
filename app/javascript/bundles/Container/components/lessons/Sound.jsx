import React, { Fragment } from "react";
import SoundPng from "images/sound.png";
import SoundGif from "images/sound.gif";
import SoundText from "./text/SoundText";

class Sound extends React.Component {
  constructor(props) {
    super(props);
    this.state = { imgSrc: SoundPng };
    this.soundGif = this.soundGif.bind(this);
    this.soundPng = this.soundPng.bind(this);
  }

  soundGif() {
    this.setState({ imgSrc: SoundGif });
  }

  soundPng() {
    this.setState({ imgSrc: SoundPng });
  }

  render() {
    const { imgSrc } = this.state;
    return (
      <Fragment>
        <section id="visual">
          <div id="nav-one" className="nav">
            <audio id="beep-one">
              <source
                src="https://cdn.rawgit.com/JessicaML/ScienceMotion-Static/aa82537e/sound_waves/wouldnt_say.mp3"
                controls
              />
              <source
                src="https://cdn.rawgit.com/JessicaML/ScienceMotion-Static/aa82537e/sound_waves/wouldnt_say.ogg"
                controls
              />
              Your browser isn't invited for super fun time.
            </audio>
            <a href="#" id="sound">
              <img
                src={imgSrc}
                onMouseOver={this.soundGif}
                onMouseOut={this.soundPng}
              />
            </a>
          </div>
        </section>
        <SoundText />
      </Fragment>
    );
  }
}
export default Sound;
