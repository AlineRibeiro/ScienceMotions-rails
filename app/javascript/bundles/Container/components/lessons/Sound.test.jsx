import React from 'react';
import renderer from 'react-test-renderer';
import Sound from './Sound';

describe('Sound', () => {
  it('renders correctly when there are no items', () => {
    const rendered = renderer.create(<Sound />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
})
