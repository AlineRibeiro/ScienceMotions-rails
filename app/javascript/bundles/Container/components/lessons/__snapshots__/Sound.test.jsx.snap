// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sound renders correctly when there are no items 1`] = `
Array [
  <section
    className="lesson-container"
    id="visual"
  >
    <div
      className="nav"
      id="nav-one"
    >
      <audio
        id="beep-one"
      >
        Your browser isn't invited for super fun time.
      </audio>
      <a
        className="sound__waveImage"
        id="sound"
        onClick={[Function]}
      >
        <img
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          src="test-file-stub"
          style={
            Object {
              "height": "175px",
              "width": "800px",
            }
          }
        />
      </a>
    </div>
  </section>,
  <section
    id="text"
  >
    <div>
      <h1>
        Properties of sound waves
      </h1>
      <p>
        Sound waves are
        <span
          id="text-1"
        >
           longitudinal
        </span>
         waves which means they vibrate in the same direction as the direction of travel. They are produced by vibrating sources, such as speakers. Sound waves can only travel through a solid, liquid or gas medium. They travel fastest in solids, then liquids and slowest in gases. A good model for sound waves is a spring.
      </p>
      <p>
        The sections of the sound waves where the particles are pushed together are areas of
        <span
          id="text-2"
        >
           compression
        </span>
        . The sections of the sound waves where the particles are further apart are areas of
        <span
          id="text-3"
        >
           rarefaction
        </span>
        .
      </p>
      <p>
        Sound waves can be
        <span
          id="text-1"
        >
           reflected
        </span>
         to form an echo,
        <span
          id="text-2"
        >
           refracted
        </span>
         and
        <span
          id="text-3"
        >
           diffracted
        </span>
        .
      </p>
    </div>
  </section>,
]
`;
